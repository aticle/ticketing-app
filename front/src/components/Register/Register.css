.login {
    padding: 20px;
    display: flex;
    align-items: center;
    flex-direction: column;
}

.login>div {
    margin: 5px 0;
    display: flex;
    width: 300px;
    max-width: 100%;
    text-align: left;
}

.login button[type="submit"] {
    margin-top: 65px;
}