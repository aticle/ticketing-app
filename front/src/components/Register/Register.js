// @flow
import React, { Component } from 'react';
import { TextField, Button } from '@material-ui/core';
import { connect, type Dispatch } from 'react-redux';
import { withRouter, RouterHistory } from 'react-router-dom';
import { registerUser } from '../../actions/authAction';
import { type User } from '../../actions/authAction';
import './Register.css';

type Props = {
    registerUser: Dispatch,
    errors: Array<Error>,
    history: RouterHistory
};
type State = {
    name: string,
    email: string,
    password: string,
    password_confirm: string,
    errors: Object
};

class Register extends Component<Props, State> {
    state = {
        name: '',
        email: '',
        password: '',
        password_confirm: '',
        errors: {}
    };

    handleChange = (name: string) => (e: SyntheticEvent<HTMLInputElement>) => {
        const value = e.currentTarget.value;

        this.setState({
            ...this.state,
            [name]: value
        });
    }

    handleSubmit = (e: Event) => {
        e.preventDefault();
        const user = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            password_confirm: this.state.password_confirm
        }
        this.props.registerUser(user, this.props.history);
    }

    render() {
        const { name, email, password, password_confirm } = this.state;

        return (
            <form className="login" onSubmit={this.handleSubmit}>
                <h2>Register</h2>
                <TextField
                    label="full name"
                    id="name"
                    type="name"
                    value={name}
                    required
                    onChange={this.handleChange('name')}
                ></TextField>
                <TextField
                    label="email"
                    id="email"
                    type="email"
                    value={email}
                    required
                    onChange={this.handleChange('email')}
                ></TextField>
                <TextField
                    label="password"
                    id="password"
                    type="password"
                    value={password}
                    required
                    onChange={this.handleChange('password')}
                ></TextField>
                <TextField
                    label="password confirm"
                    id="password_confirm"
                    type="password"
                    value={password_confirm}
                    required
                    onChange={this.handleChange('password_confirm')}
                ></TextField>
                <Button
                    id="submit"
                    variant='contained'
                    type="submit"
                >REGISTER</Button>
            </form>
        );
    };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
        registerUser: (user: User, history: RouterHistory) => registerUser(user, history)(dispatch)
    }
};

const mapStateToProps = (state: State) => {
    return {
        errors: state.errors
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Register));