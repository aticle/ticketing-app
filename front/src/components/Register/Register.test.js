import React from 'react';
import { shallow } from 'enzyme';
import Register, { mapStateToProps, mapDispatchToProps } from "./Register";

describe('Register', () => {
    it('renders correctly', () => {
        const register = shallow(<Register.WrappedComponent />);
        expect(register).toMatchSnapshot();
    });

    it('mapStateToProps correctly', () => {
        expect(mapStateToProps({ errors: {} })).toHaveProperty('errors', {});
    });

    it('mapDispatchToProps correctly', () => {
        const dispatch = jest.fn();
        expect(mapDispatchToProps(dispatch)).toHaveProperty('registerUser');
    });

    it('handles input change', () => {
        const register = shallow(<Register.WrappedComponent />);

        // name
        const nameVal = 'name';
        const nameEvent = {
            currentTarget: {
                value: nameVal
            }
        };
        register.find('#name').simulate('change', nameEvent);
        expect(register.state().name).toBe(nameVal);
    });

    it('handles submit', () => {
        const props = {
            registerUser: jest.fn(),
            history: jest.fn()
        };
        const register = shallow(<Register.WrappedComponent {...props} />);
        const event = { preventDefault: jest.fn };
        const registerInstance = register.instance();
        const user = {
            name: register.state().name,
            email: register.state().email,
            password: register.state().password,
            password_confirm: register.state().password_confirm
        };

        registerInstance.handleSubmit(event);
        expect(registerInstance.props.registerUser.mock.calls[0]).toEqual([user, registerInstance.props.history]);
    });
});
