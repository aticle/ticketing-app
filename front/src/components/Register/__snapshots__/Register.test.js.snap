// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Register
        </h2>,
        <TextField
          id="name"
          label="full name"
          onChange={[Function]}
          required={true}
          select={false}
          type="name"
          value=""
          variant="standard"
        />,
        <TextField
          id="email"
          label="email"
          onChange={[Function]}
          required={true}
          select={false}
          type="email"
          value=""
          variant="standard"
        />,
        <TextField
          id="password"
          label="password"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />,
        <TextField
          id="password_confirm"
          label="password confirm"
          onChange={[Function]}
          required={true}
          select={false}
          type="password"
          value=""
          variant="standard"
        />,
        <WithStyles(Button)
          id="submit"
          type="submit"
          variant="contained"
        >
          REGISTER
        </WithStyles(Button)>,
      ],
      "className": "login",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register",
        },
        "ref": null,
        "rendered": "Register",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "name",
          "label": "full name",
          "onChange": [Function],
          "required": true,
          "select": false,
          "type": "name",
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "email",
          "label": "email",
          "onChange": [Function],
          "required": true,
          "select": false,
          "type": "email",
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "password",
          "label": "password",
          "onChange": [Function],
          "required": true,
          "select": false,
          "type": "password",
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "password_confirm",
          "label": "password confirm",
          "onChange": [Function],
          "required": true,
          "select": false,
          "type": "password",
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "REGISTER",
          "id": "submit",
          "type": "submit",
          "variant": "contained",
        },
        "ref": null,
        "rendered": "REGISTER",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Register
          </h2>,
          <TextField
            id="name"
            label="full name"
            onChange={[Function]}
            required={true}
            select={false}
            type="name"
            value=""
            variant="standard"
          />,
          <TextField
            id="email"
            label="email"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />,
          <TextField
            id="password"
            label="password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />,
          <TextField
            id="password_confirm"
            label="password confirm"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />,
          <WithStyles(Button)
            id="submit"
            type="submit"
            variant="contained"
          >
            REGISTER
          </WithStyles(Button)>,
        ],
        "className": "login",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register",
          },
          "ref": null,
          "rendered": "Register",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "name",
            "label": "full name",
            "onChange": [Function],
            "required": true,
            "select": false,
            "type": "name",
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "email",
            "label": "email",
            "onChange": [Function],
            "required": true,
            "select": false,
            "type": "email",
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "password",
            "label": "password",
            "onChange": [Function],
            "required": true,
            "select": false,
            "type": "password",
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "password_confirm",
            "label": "password confirm",
            "onChange": [Function],
            "required": true,
            "select": false,
            "type": "password",
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "REGISTER",
            "id": "submit",
            "type": "submit",
            "variant": "contained",
          },
          "ref": null,
          "rendered": "REGISTER",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
